symplyphysics/core/experimental/points/__init__.py

```diff
@@ -0,0 +1,129 @@
+"""
+Points
+======
+
+A **point** in a 3D space is described by 3 coordinates which depend on the choice of the
+coordinate system.
+"""
+
+from __future__ import annotations
+
+from typing import Sequence, Iterator, Any
+
+from sympy import Expr
+
+from ...expr_comparisons import expr_equals
+from ..coordinate_systems import BaseCoordinateSystem
+
+
+class Point:
+    """
+    Class defining the `Point` functionality.
+    """
+
+    _coordinates: tuple[Expr, Expr, Expr]
+    _system: BaseCoordinateSystem
+
+    @property
+    def coordinates(self) -> tuple[Expr, Expr, Expr]:
+        """
+        Coordinates of the point.
+        """
+
+        return self._coordinates
+
+    @property
+    def system(self) -> BaseCoordinateSystem:
+        """
+        Coordinate system the point is attached to.
+        """
+
+        return self._system
+
+    def __init__(
+        self,
+        coordinates: Sequence[Expr],
+        system: BaseCoordinateSystem,
```

Comment:
I wanted to avoid default parameters and make everything explicit. It only adds one line in the actual code (e.g. `c = CartesianCoordinateSystem("C")`) but the following code will be clear about where and which coordinate system is used.

Reply:
Most people do not care about coordinate systems. They want just a point with 2 or 3 coordinates.